rootfs/etc/profile.d/aws-vault.sh

```diff
@@ -67,8 +95,16 @@ if [ "${AWS_VAULT_ENABLED:-true}" == "true" ]; then
 	function aws_vault_assume_role() {
 		# Do not allow nested roles
 		if [ -n "${AWS_VAULT}" ]; then
-			echo "Type '$(green exit)' before attempting to assume another role"
-			return 1
+			if [ "$SHLVL" -eq 1 ] && [ "${AWS_VAULT_SERVER_ENABLED:-true}" == "true" ]; then
+				local aws_vault="$AWS_VAULT"
+				local aws_vault_server_enabled="${AWS_VAULT_SERVER_ENABLED:-true}"
+				trap 'export AWS_VAULT="$aws_vault" && export AWS_VAULT_SERVER_ENABLED="$aws_vault_server_enabled" && trap - RETURN' RETURN
```

Comment:
Can you add a shell comment above this that explains the intent of what this code does? E.g. why the trap and nested trap?

Reply:
Done